sdk/core/azure-core/inc/azure/core/internal/testing_macro.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief This file is used to define internal-only macros that are used to control the behavior of
+ * the Azure SDK when running tests, to allow mocking within tests.
+ * The macros in this file should NOT be used by anyone outside this repo.
+ */
+
+#pragma once
+
+// When testing is enabled, we want to make sure that certain classes are not final, so that we can
+// mock it.
+#if defined(_azure_TESTING_BUILD)
```

Comment:
AMQP had its own `_azure_TESTING_BUILD_AMQP` for some reason. I think it should be renamed `_azure_TESTING_BUILD_AMQP` => `_azure_TESTING_BUILD`, and these macros will work as expected in all contexts.

Reply:
Will address separately https://github.com/Azure/azure-sdk-for-cpp/issues/5402